pkg/storage/replica.go

```diff
@@ -3653,30 +3639,30 @@ func (r *Replica) processRaftCommand(
 			// Close over raftCmd to capture its value at execution time; we clear
 			// ReplicatedEvalResult on certain errors.
 			defer func() {
-				splitMergeUnlock(*raftCmd.ReplicatedEvalResult)
+				splitMergeUnlock(raftCmd.ReplicatedEvalResult)
 			}()
 		}
 	}
 
 	var response proposalResult
 	var writeBatch *storagebase.WriteBatch
 	{
-		if raftCmd.ReplicatedEvalResult == nil && forcedErr == nil {
+		if reflect.DeepEqual(raftCmd.ReplicatedEvalResult, storagebase.ReplicatedEvalResult{}) && forcedErr == nil {
```

Comment:
I really don't think there should be `reflect` in this code path. You could have `(*ReplicatedEvalResult).Empty() bool` with a corresponding test that doesn't rot when new fields are added.